clang/include/clang/Basic/Attr.td

```diff
@@ -2794,11 +2789,6 @@ def ReqdWorkGroupSize : InheritableAttr {
     ArrayRef<const Expr *> dimensions() const {
       return {getXDim(), getYDim(), getZDim()};
     }
-    bool isValueDependent() const {
-      return (getXDim() && getXDim()->isValueDependent()) ||
-             (getYDim() && getYDim()->isValueDependent()) ||
-             (getZDim() && getZDim()->isValueDependent());
-    }
```

Comment:
I have removed this from the current patch since it is not being used and any of (XYZ)Dim is value dependent, we give diagnostics and tests are added to support for this.